crates/hir/src/lib.rs

```diff
@@ -1418,7 +1418,8 @@ impl Adt {
     }
 
     pub fn layout(self, db: &dyn HirDatabase) -> Result<Layout, LayoutError> {
-        if db.generic_params(self.into()).iter().count() != 0 {
+        let generic_params = &db.generic_params(self.into());
+        if generic_params.iter().next().is_some() || generic_params.iter_lt().next().is_some() {
```

Comment:
Having a `GenericParam::is_empty` method would probably be useful here (or given there is more than just params in it, a `GenericParam::has_params`)

Reply:
Should I open a PR with the proposed change of this and the proposed change in another PR?

Feel like a opening a lot of small PRs 😅.

Reply:
Putting both into one is fine